pkg/cvo/updatepayload.go

```diff
@@ -304,13 +304,21 @@ func findUpdateFromConfig(config *configv1.ClusterVersion) (configv1.Update, boo
 
 func findUpdateFromConfigVersion(config *configv1.ClusterVersion, version string, force bool) (configv1.Update, bool) {
 	for _, update := range config.Status.AvailableUpdates {
-		if update.Version == version {
-			return update, len(update.Image) > 0
+		if update.Version == version && len(update.Image) > 0 {
```

Comment:
So the semantics have changed -> if `Image` is not set do not return `Version`? And same applies for history check below.

Reply:
yeah.  Before if some crazy upstream service had declared a node with a matching version but no image, we'd have said "sorry, no match" without looking through history.  Now we say "skipping this useless entry" and move on to consider later entries and history.  Documented in the commit message with 1316b58c -> 33333ecb.